src/SymfonyCorp/Connect/Api/Api.php

```diff
@@ -32,15 +34,22 @@ class Api
 {
     const ENDPOINT = 'https://connect.symfony.com/api';
 
-    private $browser;
+    private $httpClient;
     private $parser;
     private $logger;
     private $endpoint;
     private $accessToken;
 
-    public function __construct($endpoint = null, Browser $browser = null, ParserInterface $parser = null, LoggerInterface $logger = null)
+    public function __construct($endpoint = null, HttpClientInterface $httpClient = null, ParserInterface $parser = null, LoggerInterface $logger = null)
     {
-        $this->browser = $browser ?: new Browser(new Curl());
+        if (class_exists(Browser::class) && $httpClient instanceof Browser) {
+            @trigger_error(sprintf('Passing a "%s" to "%s()" is deprecated since 5.1, use "%s" instead.', Browser::class, __METHOD__, HttpClientInterface::class), E_USER_DEPRECATED);
```

Comment:
we must always mention the package that is triggering the deprecation:
...since symfony/connect 5.1...